dipy/denoise/noise_estimate.py

```diff
@@ -274,6 +296,23 @@ def estimate_sigma(arr, disable_background_masking=False):
     k[1, 1, 0] = 1
     k[1, 1, 2] = 1
 
+    # Precomputed factor from Koay 2006, this corrects the bias of magnitude image
+    correction_factor = {1: 0.42920367320510366,
+                         4: 0.4834941393603609,
+                         6: 0.4891759468548269,
+                         8: 0.49195420135894175,
+                        12: 0.4946862482541263,
+                        16: 0.4960339908122364,
+                        20: 0.4968365823718557,
+                        24: 0.49736907650825657,
+                        32: 0.49803177052530145,
+                        64: 0.49901964176235936}
+
+    if N in correction_factor:
+        factor = correction_factor[N]
+    else:
+        factor = correction_factor[1]
```

Comment:
Just to be clear: I believe that the 'default to 1' part is already taken
care of.

On Tue, May 19, 2015 at 8:44 AM, Ariel Rokem <arokem@gmail.com> wrote:

> Exactly: raise an error if a number is provided, but it is not a key to
> the `correction_factor` dict. Otherwise, default to 1.
>
> I propose to rewrite this logic as:
>
>     if N in correction_factor:
>         factor = correction_factor[N]  # Per default, N = 1
>     else:
>        raise ValueError("Correction factor needs to be one of the
> following: %s, but you entered %s" %(list(correction_factor.keys()),  N))
>
> With the necessary line breaks, etc.
>
> On Tue, May 19, 2015 at 8:37 AM, Samuel St-Jean <notifications@github.com>
> wrote:
>
>> In dipy/denoise/noise_estimate.py
>> <https://github.com/nipy/dipy/pull/651#discussion_r30612485>:
>>
>> > +    # Precomputed factor from Koay 2006, this corrects the bias of magnitude image
>> > +    correction_factor = {1: 0.42920367320510366,
>> > +                         4: 0.4834941393603609,
>> > +                         6: 0.4891759468548269,
>> > +                         8: 0.49195420135894175,
>> > +                        12: 0.4946862482541263,
>> > +                        16: 0.4960339908122364,
>> > +                        20: 0.4968365823718557,
>> > +                        24: 0.49736907650825657,
>> > +                        32: 0.49803177052530145,
>> > +                        64: 0.49901964176235936}
>> > +
>> > +    if N in correction_factor:
>> > +        factor = correction_factor[N]
>> > +    else:
>> > +        factor = correction_factor[1]
>>
>> No I am indeed forcing a value. I just precomputed a bunch of mostly
>> plausible values. What I'm not sure of though is what to do if the user
>> does not input any value for N. All the old code won't be working anymore,
>> but I could raise an error if the number is not in the list, 1 for no
>> input, and the value as requested if it's in the list.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/nipy/dipy/pull/651/files#r30612485>.
>>
>
>